### PR TITLE
feat: add https capabilities to dev server by adding `https` in nuxt config.

### PR DESCRIPTION
**Pull request by achretien**
The parameter could be `true` or options from 

https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener

### CODE REVIEW

**Comment by Alex94800** [comment]
+1

**Review by atinux — CHANGES_REQUESTED** [review]
Agree about this PR, could you add a test about it please?

**Comment by Atinux** [comment]
You can add a test file called `test/unit/https.test.js` and take some inspiration from https://github.com/nuxt/nuxt.js/blob/dev/test/unit/express.test.js

The point is to test that it renders correctly on the https url and not http. And also to verify that the coverage does not decrease.

Make sure to run `npm run test:fixtures` before running `npx jest test/unit/https.test.js`.

Inside your test, you can do something like this:

```js
 const config = loadFixture('basic')
nuxt = new Nuxt(config)
port = await getPort()
await nuxt.listen(port, '0.0.0.0', /* your https options */)
```

File: bin/nuxt-dev [hunk 1]
```diff
@@ -111,7 +111,7 @@ function startDev(oldInstance) {
         if (oldInstance && oldInstance.builder) {
           return oldInstance.builder.unwatch()
         } else {
-          return nuxt.listen(port, host)
+          return nuxt.listen(port, host, nuxtConfig.https)
```
Comment (clarkdo) [inline bin/nuxt-dev]: Maybe unnecessary to pass `https`, config can be got through `this.options` inside `nuxt.js`

**Review by clarkdo — CHANGES_REQUESTED** [review]
This feature sounds great to me, thank you for the contribution @achretien 

In my perspective, it may be not that important to add `https` in `nuxt dev`, since it's designed for development, so maybe it can be introduced in `nuxt start`.

And also, it will be better if add default value of `https` into `common/nuxt.config.js`

**Comment by zyxep** [comment]
Why not give the developer option to run https on `nuxt dev`.
a case could be: using fb login.
Their developer platform "require" https callback urls.

some tools give devs options to create cert to e.g. localhost with e.g.
https://github.com/FiloSottile/mkcert

**Comment by galvez** [comment]
Closing this as I'll push a separate forked branch from this that integrates with https://github.com/nuxt/nuxt.js/pull/3812.

**Pull request closed without merging**